nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -424,6 +440,35 @@ void DockingServer::doInitialPerception(Dock * dock, geometry_msgs::msg::PoseSta
   }
 }
 
+void DockingServer::rotateToDock(const geometry_msgs::msg::PoseStamped & dock_pose)
+{
+  rclcpp::Rate loop_rate(controller_frequency_);
+  const double dt = 1.0 / controller_frequency_;
+  auto target_pose = dock_pose;
+  target_pose.pose.orientation = nav2_util::geometry_utils::orientationAroundZAxis(
+    tf2::getYaw(target_pose.pose.orientation) + M_PI);
+
+  while (rclcpp::ok()) {
```

Comment:
Introduce a `rotate_to_dock_timeout` parameter to compliment the other timeouts for while loops. While loops scare me bc they're infinite so I want a fallback to get out of them after a generous time period in case a problem occurs